go/libraries/doltcore/schema/alterschema/droppk_test.go

```diff
@@ -124,3 +124,134 @@ func TestDropPk(t *testing.T) {
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (age))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if compound backup index for compound FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `backup` (id, age))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id, age) references parent (id, age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "no error if both invalid and valid backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id, age), key `bad_backup1` (id, age), key `bad_backup2` (id), key `backup` (age, id))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (age) references parent (age))",
+			},
+			exit:      0,
+			fkIdxName: "backup",
+		},
+		{
+			name: "error if FK ref but no backup index for single pk",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id))",
+				"create table child (id int, name varchar(1), age int, primary key (id), constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if FK ref but bad backup index",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, primary key (id), key `bad_backup2` (age))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if misordered compound backup index for FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, constraint `primary` primary key (id), key `backup` (age, id))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (id) references parent (id))",
+			},
+			exit:      1,
+			fkIdxName: "id",
+		},
+		{
+			name: "error if incomplete compound backup index for FK",
+			setup: []string{
+				"create table parent (id int, name varchar(1), age int, constraint `primary` primary key (age, id), key `backup` (age, name))",
+				"create table child (id int, name varchar(1), age int, constraint `fk` foreign key (age, id) references parent (age,  id))",
+			},
+			exit:      1,
+			fkIdxName: "ageid",
+		},
+	}
+
+	for _, tt := range dropTests {
+		childName := "child"
+		parentName := "parent"
+		childFkName := "fk"
+
+		t.Run(tt.name, func(t *testing.T) {
+			dEnv := dtestutils.CreateTestEnv()
+			ctx := context.Background()
+			for _, c := range tt.setup {
+				exitCode := commands.SqlCmd{}.Exec(ctx, "sql", []string{fmt.Sprintf("-q %s", c)}, dEnv)
```

Comment:
This is a really cumbersome way to execute SQL statements. Why don't you use the SQL test utility functions here?

Reply:
i agree i hate this, which utility functions? when i asked last week you pointed me towards enginetests